### PR TITLE
fix(box): ensure z-index is set when sticky or fixed positioned

### LINKED ISSUE #5572: inputs overlap fixed positions
### Current behaviour

All kind of inputs (radio buttons, textbox, select...) overlap a box with the `position: sticky` rule, as they have a z-index of 999, while the box doesn't have a custom z-index.

image

### Expected behaviour

For the inputs to go under the sticky box, as the rest of the text.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/sidebar-with-sticky-content-forked-xgyefg?file=/src/index.js

### JIRA Ticket (Sage Only)

SBS-44457

### Suggested Solution

_No response_

### Carbon Version

111.4.2

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by edleeks87**
fix #5572

### Proposed behaviour

Ensures that the any `Box` container that has `sticky` or `fixed` positioning has a z-index set to prevent it being overlaid by other elements that are relatively positioned

### Current behaviour

Any `Box` container that has `sticky` or `fixed` positioning has no z-index set

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

See sandbox in linked issue below

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/box/box.spec.tsx [hunk 1]
```diff
@@ -119,4 +120,17 @@ describe("Box", () => {
       wrapper
     );
   });
+
+  it.each<BoxProps["position"]>(["fixed", "sticky"])(
+    "sets zIndex to 1 when position is %s",
```
Comment (DipperTheDan) [inline src/components/box/box.spec.tsx:125]: Is it worth doing a test that covers the undefined path of this too? Might be overkill but I figured I'd ask the question anyway.

**Review by Parsium — COMMENTED** [review]
Looks good to me 👍🏼 Happy to accept once @DipperTheDan's query has been addressed :)

File: src/components/box/box.spec.tsx (hunk 1 above)
Comment (edleeks87) [reply src/components/box/box.spec.tsx:125]: I feel it's somewhat redundant to assert the zIndex is undefined when the position is undefined if that's what you mean?

Comment (DipperTheDan) [reply src/components/box/box.spec.tsx:125]: Not quite. I was referring to if the position css value is `absolute` for example then asserting that the zIndex is `undefined`. Just something that explores the `false` path of the ternary statement.

**Review by DipperTheDan — APPROVED** [review]

File: src/components/box/box.spec.tsx (hunk 1 above)
Comment (edleeks87) [reply src/components/box/box.spec.tsx:125]: Those paths are implicitly covered in other test cases but I think there's some value in having an explicit assertion so will add

**Review by Parsium — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**